doc/plantuml/provisioning_operation_detailed_design.pu

```diff
@@ -0,0 +1,102 @@
+
+provisioning -> mqtt: Subscribe to 'accepted' and 'rejected' response topics of operation
+note right of provisioning: Register MQTT subscription callback\nfor processing server response.
+
+mqtt -> aws: Transmit MQTT SUBSCRIBE packets to Server
+mqtt -> provisioning: Return SUCCESS or error
+provisioning --> app: Return MQTT_ERROR if error in subscribing
+
+provisioning -> serializer: Calculate size of operation request payload\nfor PUBLISH message
+serializer -> serializer: Performs <i>dry-run serialization</i>\nfor calculating payload size
+serializer -> provisioning: Return calculated size OR error for failure.
+provisioning -> app: Return INTERNAL_FAILURE if calculation failed.
+
+provisioning -> provisioning: Allocate memory for payload buffer
+provisioning -> app: Return NO_MEMORY if allocation fails.
+
+provisioning -> serializer: Request serialization of PUBLISH payload. \nProvide allocated buffer.  
+serializer -> serializer: Perform <i>actual serialization</i> in buffer.
+serializer -> provisioning: Return SUCCESS or INTERNAL_FAILURE/NO_MEMORY\nfor serializer failure.
+provisioning -> app: Return same error code if serializalion failed.
+
+provisioning -> mqtt: PUBLISH payload to operation's request topic
+mqtt -> aws: Transmit MQTT PUBLISH packets to server
+mqtt -> provisioning: Return SUCCESS or failure.
+provisioning --> app: Return MQTT_ERROR if PUBLISH failed.
+
+provisioning -> provisioning: Wait on server response with timeout.
+note right of provisioning: <b>Sempahore</b> resource used for pending on\n notification from MQTT subscription callback.
+alt Receive Server Response within timeout period
+deactivate app
+aws -> mqtt : Incoming PUBLISH response from server
+mqtt -> callback: Invoke MQTT subscription callback.\n Pass server response information.
+activate callback
+callback -> callback: Parse incoming PUBLISH topic for\nACCEPTED/REJECTED suffix to determine\n type of server response.
+callback -> parser: Invoke response payload parser of operation.
+note left of callback: Pass accepted OR rejected response type,\nresponse payload, user's response callback.
+parser -> parser: Parse response payload
+alt If error in parsing
+parser -> callback: Return appropriate error (BAD_RESPONSE, INTERNAL_FAILURE)
+else Successful Payload Parsing
+parser -> userCallback: Invoke user-callback.\nPass parsed payload information.
+parser -> callback: Return SUCCESS
+end alt
+callback -> provisioning: Notify pending API calling context about server response reception
+note left of callback: Notification done on the shared <b>semaphore resource</b>.
+alt Corner Case: Server response received after Provisionin library cleanup
+callback -> callback: Destroy semaphore resource
+note right of callback: Tackles <b>RACE CONDITION</b> between receiving\nduplicate server responses (MQTT QoS 1)\nand cleaning library after API returns.
+end alt
+deactivate callback
+activate app
```

Comment:
This seems to be not working in the diagram generated. It is better to have the task context shown

Reply:
Good observation. Made minor section movements to fix the rendering issue